autoload/codefmt/zprint.vim

```diff
@@ -50,33 +76,33 @@ function! codefmt#zprint#GetFormatter() abort
   " @flag(zprint_executable), only targeting the range between {startline} and
   " {endline}.
   function l:formatter.FormatRange(startline, endline) abort
-    " Must be upper-cased to call as a function
-    let l:ZprintOptions = s:plugin.Flag('zprint_options')
-    if type(l:ZprintOptions) is# type([])
-      " Assign upper-case to lower-case
-      let l:zprint_options = l:ZprintOptions
-    elseif maktaba#value#IsCallable(l:ZprintOptions)
-      " Call upper-case to assign lower-case
-      let l:zprint_options = maktaba#function#Call(l:ZprintOptions)
-    else
-      throw maktaba#error#WrongType(
-            \ 'zprint_options flag must be list or callable. Found %s',
-            \ string(l:ZprintOptions))
+    let l:exe = codefmt#formatterhelpers#ResolveFlagToArray(
+          \ 'zprint_executable')
+    let l:opts = codefmt#formatterhelpers#ResolveFlagToArray(
+          \ 'zprint_options')
+
+    if s:plugin.Flag('apply_vim_settings')
```

Comment:
A global flag won't support the per-path or per-language behavior this ought to support. Even if it's easy to disable it globally, you can now get into a position where you have a single case it doesn't work well for and you have this unhelpful choice between using it everywhere or using it nowhere.

Reply:
Will address in separate PR